.github/workflows/deploy.yml

```diff
@@ -1,15 +1,15 @@
-name: Deploy
+name: Reusable Deploy Workflow
 
 on:
-  workflow_dispatch:
-  schedule:
-    - cron: "0 3 * * *" # Every night at 03:00 AM
-  push:
-    branches: ["main"]
+  workflow_call:
+    inputs:
+      profile:
+        required: true
+        type: string
```

Comment:
I am unsure if this works regarding secrets. [The GitHub documentation on reusable workflows suggests additional fields to be specified.](https://docs.github.com/en/enterprise-cloud@latest/actions/how-tos/reuse-automations/reuse-workflows#using-inputs-and-secrets-in-a-reusable-workflow)